### ISSUE TITLE
"invalid memory reference" in Racket CS 

### ISSUE THREAD

**Issue by alex-hhh** [racket/racket#3753]
**What version of Racket are you using?**

8.0.0.12 [cs] (snapshot from 31st March 2021, downloaded from the Utah servers), running on Windows 10

DrRacket prints out: Welcome to DrRacket, version 8.0.0.12--2021-03-31(9a4f968/a) [cs].

**What program did you run?**

[[This is the simplest set of steps I could find, that reproduces the problem, the problem happens with all charts in AL2, but not in other simpler applications which make use of the "plot hover" functionality.]]

* clone and build https://github.com/alex-hhh/ActivityLog2
* run the application using "racket run.rkt"
* create a new database and import an activity for example the file from "tests/test-data/920xt-run.fit"
* open the session the application, go to the charts tab and move the mouse over the plot

**What should have happened?**

Additional information is displayed over the plots when the mouse is moved over them, instead an "invalid memory reference error" message is printed and the application hangs.

**If you got an error message, please include it here.**

```
invalid memory reference.  Some debugging context lost
  context...:
   C:\Program Files\Racket\share\pkgs\plot-gui-lib\plot\private\gui\snip2d.rkt:204:4: draw-overlay-renderers
   C:\Program Files\Racket\share\pkgs\gui-lib\mred\private\wxme\pasteboard.rkt:1213:2: draw method in pasteboard%
   C:\Program Files\Racket\share\pkgs\gui-lib\mred\private\wxme\pasteboard.rkt:1318:2: refresh method in pasteboard%
   C:\Program Files\Racket\share\pkgs\gui-lib\mred\private\wxme\editor-canvas.rkt:648:2: redraw method in editor-canvas%
   C:\Program Files\Racket\share\pkgs\gui-lib\mred\private\wxme\editor-canvas.rkt:1294:4: all-in-chain method in canvas-editor-admin%
   C:\Program Files\Racket\collects\racket\contract\private\arrow-val-first.rkt:486:18
   C:\Users\alexh\Projects\ActivityLog2\rkt\session-inspector\inspect-graphs.rkt:1740:4: hover-callback
   C:\Program Files\Racket\collects\racket\contract\private\arrow-higher-order.rkt:375:33
   C:\Program Files\Racket\share\pkgs\plot-gui-lib\plot\private\gui\snip2d.rkt:268:4: on-event method in 2d-plot-snip%
   C:\Program Files\Racket\share\pkgs\gui-lib\mred\private\wxme\pasteboard.rkt:250:2: on-event method in pasteboard%
   C:\Program Files\Racket\share\pkgs\gui-lib\mred\private\wxme\editor-canvas.rkt:412:2: on-event method in editor-canvas%
   C:\Program Files\Racket\collects\racket\private\more-scheme.rkt:148:2: call-with-break-parameterization
   C:\Program Files\Racket\collects\ffi\unsafe\atomic.rkt:73:13
   C:\Program Files\Racket\share\pkgs\gui-lib\mred\private\wx\win32\window.rkt:820:2: dispatch-on-event method in window%
   C:\Program Files\Racket\share\pkgs\gui-lib\mred\private\wx\common\queue.rkt:435:6
   C:\Program Files\Racket\share\pkgs\gui-lib\mred\private\wx\common\queue.rkt:486:32
```

**Please include any other relevant details**
e.g., the operating system used or how you are running the code.

**Comment by mflatt** [comment]
The crash happens at `plot-area` in "plot-lib/plot/private/no-gui/plot2d-utils.rkt" because it receives a bad argument. The type on `plot-area` says that it takes a `(Listof renderer2d)`, but it's being given a list that contains `#f`. It looks like there's no check on the argument because `plot-area` is exported with `provide-unsafe`.

**Comment by samth** [comment]
Can you try racket/plot#91 and see if that fixes it?

**Comment by mflatt** [comment]
I think this was fixed in racket/plot@9189c8f71d0585ea46a82f9032e9e706243417eb. Ok to close the issue?

**Comment by alex-hhh** [comment]
Yes, this can be closed.
